config/main.py

```diff
@@ -984,18 +1033,235 @@ def remove(ctx, interface_name, ip_addr):
 
     try:
         ipaddress.ip_network(unicode(ip_addr), strict=False)
-        if interface_name.startswith("Ethernet"):
-            config_db.set_entry("INTERFACE", (interface_name, ip_addr), None)
-        elif interface_name.startswith("PortChannel"):
-            config_db.set_entry("PORTCHANNEL_INTERFACE", (interface_name, ip_addr), None)
-        elif interface_name.startswith("Vlan"):
-            config_db.set_entry("VLAN_INTERFACE", (interface_name, ip_addr), None)
-        elif interface_name.startswith("Loopback"):
-            config_db.set_entry("LOOPBACK_INTERFACE", (interface_name, ip_addr), None)
-        else:
+        table_name = get_interface_table_name(interface_name)
+        if table_name == "":
             ctx.fail("'interface_name' is not valid. Valid names [Ethernet/PortChannel/Vlan/Loopback]")
+        config_db.set_entry(table_name, (interface_name, ip_addr), None)
+        interface_dependent = interface_ipaddr_dependent_on_interface(config_db, interface_name)
+        if len(interface_dependent) == 0 and is_interface_bind_to_vrf(config_db, interface_name) is False:
+            config_db.set_entry(table_name, interface_name, None)
     except ValueError:
         ctx.fail("'ip_addr' is not valid.")
+
+#
+# 'vrf' subgroup ('config interface vrf ...')
+#
+
+
+@interface.group()
+@click.pass_context
+def vrf(ctx):
+    """Bind or unbind VRF"""
+    pass
+
+#
+# 'bind' subcommand
+#
+@vrf.command()
+@click.argument('interface_name', metavar='<interface_name>', required=True)
+@click.argument('vrf_name', metavar='<vrf_name>', required=True)
+@click.pass_context
+def bind(ctx, interface_name, vrf_name):
+    """Bind the interface to VRF"""
+    config_db = ctx.obj["config_db"]
+    if get_interface_naming_mode() == "alias":
+        interface_name = interface_alias_to_name(interface_name)
+        if interface_name is None:
+            ctx.fail("'interface_name' is None!")
+
+    table_name = get_interface_table_name(interface_name)
+    if table_name == "":
+        ctx.fail("'interface_name' is not valid. Valid names [Ethernet/PortChannel/Vlan/Loopback]")
+    if is_interface_bind_to_vrf(config_db, interface_name) is True and \
+        config_db.get_entry(table_name, interface_name).get('vrf_name') == vrf_name:
+        return
+    interface_dependent = interface_ipaddr_dependent_on_interface(config_db, interface_name)
+    for interface_del in interface_dependent:
```

Comment:
Looks like we're deleting all the IP addresses present on the interface here, please add comment here.

Reply:
OK, I will add later.